Source/core/NetworkInfo.h

```diff
@@ -136,6 +136,7 @@ namespace Core {
             virtual ~INotification() = default;
 
             virtual void Event(const string&) = 0;
+            virtual void AddressEvent(const string&) {}
```

Comment:
If we are going to specialize the interface lets do it explicitly. Why not :
virtual void Added(const string& interface, const Core::IPNode& address);
virtual void Removed(const string& interface, const Core::IPNode& address);
That prevents the receiver from finding out what address change happenend and you do have that information from the calling position.